koku/masu/api/report_data.py

```diff
@@ -17,20 +17,66 @@
 from rest_framework.settings import api_settings
 
 from api.models import Provider
+from api.provider.models import Provider
 from api.utils import get_months_in_date_range
+from koku.cache import get_cached_resummarize_by_provider_type
+from koku.cache import set_cached_resummarize_by_provider_type
+from masu.database.koku_database_access import KokuDBAccess
 from masu.database.provider_db_accessor import ProviderDBAccessor
 from masu.processor import is_customer_large
 from masu.processor.tasks import PRIORITY_QUEUE
 from masu.processor.tasks import PRIORITY_QUEUE_XL
 from masu.processor.tasks import QUEUE_LIST
 from masu.processor.tasks import remove_expired_data
-from masu.processor.tasks import update_all_summary_tables
 from masu.processor.tasks import update_summary_tables
+from masu.processor.tasks import update_summary_tables_by_provider
 
 LOG = logging.getLogger(__name__)
 REPORT_DATA_KEY = "Report Data Task IDs"
 
 
+class GetOpenshiftOnCloudProviders(KokuDBAccess):
+    """Class to interact with the koku database for Provider Data."""
```

Comment:
This feels inaccurate. :confused: Or I think this class does a little more than just interact with the DB :smile:

Reply:
Ill probably just back these changes out. I added them in last minute cause I thought they would help with a customer issue.